python/sparkdl/transformers/param.py

```diff
@@ -107,18 +159,74 @@ def toStringOrTFTensor(value):
                 raise TypeError("Could not convert %s to tensorflow.Tensor or str" % type(value))
 
     @staticmethod
-    def toTFGraph(value):
-        # TODO: we may want to support tf.GraphDef in the future instead of tf.Graph since user
-        # is less likely to mess up using GraphDef vs Graph (e.g. constants vs variables).
-        if isinstance(value, tf.Graph):
-            return value
-        else:
-            raise TypeError("Could not convert %s to tensorflow.Graph type" % type(value))
-
-    @staticmethod
     def supportedNameConverter(supportedList):
         def converter(value):
             if value in supportedList:
                 return value
             else:
                 raise TypeError("%s %s is not in the supported list." % type(value), str(value))
+        return converter
+
+# New in sparkdl
+
+class HasOutputMapping(Params):
+    """
+    Mixin for param outputMapping: ordered list of ('outputTensorName', 'outputColName') pairs
+    """
+    outputMapping = Param(Params._dummy(), "outputMapping",
+                          "Mapping output :class:`tf.Tensor` objects to DataFrame column names",
+                          typeConverter=SparkDLTypeConverters.asTensorToColumnMap)
+
+    def setOutputMapping(self, value):
+        return self._set(outputMapping=value)
+
+    def getOutputMapping(self):
+        return self.getOrDefault(self.outputMapping)
+
+
+class HasInputMapping(Params):
+    """
+    Mixin for param inputMapping: ordered list of ('inputColName', 'inputTensorName') pairs
+    """
+    inputMapping = Param(Params._dummy(), "inputMapping",
+                         "Mapping input DataFrame column names to :class:`tf.Tensor` objects",
```

Comment:
I will document the actual converted type here and indicate what types are automatically converted.

Reply:
My comment was, the user should give a mapping {dataframe column name : tensor name} not {data frame column name : tensor object} but the documentation says otherwise. They don't need to know about internal conversions.